src/utils/type-transform/Payload.ts

```diff
@@ -46,26 +52,55 @@ function deconstructAlpha41(str: string, throws = false): null | never {
 }
 
 const versions = new Set([deconstructAlpha40_Or_Alpha39, deconstructAlpha41])
-export function deconstructPayload(str: string): Payload | null
-export function deconstructPayload(str: string, throws: true): Payload
-export function deconstructPayload(str: string, throws = false): Payload | null {
-    for (const ver of versions) {
-        const result = ver(str, false)
-        if (throws === false) return result
-        if (result) return result
-        return ver(str, true)
+
+/**
+ * @type Decoder    defines decoder. if null, auto select decoder.
```

Comment:
Why we have a default decoder but not a default encoder?

Reply:
We do not have a default decoder (in this file). When it's not defined, the method tries every possible decoder until found working one. But every encoder will working. And we have defaults for encoder and decoder in social media config.